### ISSUE TITLE
v2.0.0 Testing

### ISSUE THREAD

**Issue by MrJohnZoidberg** [patzly/grocy-android#266]
Hello guys,
I still need some days (or maybe even weeks) to finish work on `v2.0.0` of our app. Sorry for that...
Currently it's only me who is working on it, that's a bit slow but it's easier to create consistent code.

You will get great improvements regarding
- compatibility with grocy 3.0.0 and higher
- speed
- navigation
- consistency
- download data saving (load only outdated data)
- help texts
- settings
- product editing
- shopping mode
- offline usage (no transactions but viewing lists)
- code readability
- list diff animations
- bottom app bar animations
- network requests security

And personally, I'm learning a lot about app architecture while working on this app. That's good, too!
After I finished the main work on this version, we need some testers who test it to find bugs and crashes. We will upload the alpha and beta versions as APK in this issue and you can install them on your device. When we can't find new bugs anymore (together with you) we can release it.

**So if you want to test the new version when it's ready, you can subscribe to this issue now (with the button on the right) to get notifications. I will write a comment here as soon as it is time for testing. Thanks a lot for your help!**
**You can give a thumb up so that we can see how many of you want to test. :-)**

---
We have no website where we could post such a message like this one, but I think this is enough to reach a few of you. We could also use the Testing release in Google Play Store but currently we don't want to invest time in that.

Strings will be ready for translation on Transifex as soon as I merge the dev branch into master branch. Maybe in a few days :-)

---
UPDATE 1 (01/11/21): I still need one, two or three weeks – it's more work for 3.x server compatibility than I thought...
UPDATE 2 (01/26/21): Still working on it, first alpha will come soon...
UPDATE 3 (02/01/21): I merged the 'dev' branch into master. New strings are available for translation now.
UPDATE 4 (02/08/21):
After many night sessions, we are happy to announce that later today the first alpha of Grocy Android v2.0.0 will be available here!
**Reminder: The update requires grocy v3.0.x on your server, so now is the time to update if you haven't already.**
Till then! _~patzly_

**Comment by firechhe** [comment]
#subscribe

**Comment by Gerw88** [comment]
I'm in 👍

**Comment by SamBouwer** [comment]
I'm also in!

**Comment by towo** [comment]
Please just use the subscribe button on the right, otherwise you'll just be
spamming everyone also subscribed.

**Comment by hansidampf** [comment]
I could also help testing.

**Comment by kenguest** [comment]
I'm in!

**Comment by samnewman86** [comment]
Hi, where is the link to the APK? I'm happy to test with server 3.0

**Comment by wilfredktlr** [comment]
> Hi, where is the link to the APK? I'm happy to test with server 3.0

Let's quote the author...

> After many night sessions, we are happy to announce that **later today** the first alpha of Grocy Android v2.0.0 will be available **here**!

**Comment by samnewman86** [comment]
Sorry, I came in and saw that and read the date as yesterday

**Comment by MrJohnZoidberg** [comment]
Thank you that so many of you want to test the new version while it's in alpha/beta stage!

**TL;DR: Don't install this version if you want a fully working app, some pages or features are not complete yet – especially actions on the stock overview page are not implemented yet. You need grocy v3.0.0 or v3.0.1 on your server. And don't install this version if you have Due date tracking or Location tracking feature disabled on your server, else there may be crashes/misbehavior! For more details read the text below.**

APK: [Grocy-Android-v2.0.0-alpha01.zip](https://github.com/patzly/grocy-android/files/5946299/Grocy-Android-v2.0.0-alpha01.zip) (you have to unzip this file)

Here is a changelog for this first alpha:

**Compatibility**
- Compatible with grocy v3.0.x
- Not compatible anymore with grocy v2.7.x

**General improvements**
- Improved page loading speed by use of database
- Less data usage by use of database (only fetch new data if necessary)
- Offline usage (no transactions but viewing lists)
- List difference animations are used in every list
- Smooth & consistent navigation between pages
- Product autocomplete lists now use fuzzy finder (idea from grocy/grocy#1275)
- Server requests are now more secure with API key in header
- Added many help texts which are also in the web interface (e.g. on master product pages)
- No more buggy bottom app bar animations
- Fixed layout for text on features pages
- Rewritten font usage and styling system

**New languages (Thanks for your efforts!)**
- Russian
- Ukrainian

**New pages**
- Overview
- Product create/edit
- External scanner

**Removed pages**
- Batch mode (consume & purchase forms now behave like web interface and have a scan mode instead)
- Missing batch items (unnecessary without batch mode)

**Rewritten & strongly edited pages**
- Purchase
- Consume
- Settings (with lots of new options)
- Login (New login flow for QR codes from grocy server)

**:warning:Not fully working yet (coming in later test versions):warning:**
Please don't report issues yet for these points!
- Purchase action from shopping list not implemented yet
- Stock overview (actions not working yet)
- Product page (barcodes and unit conversions functionality not implemented yet)
- Settings page (not all options implemented yet)
- Tor and SOCKS support (not implemented yet)
- **Deactivated features: if Due Date Tracking or Location Tracking feature is disabled, there can be misbehavior!**
- Online->offline & offline->online changes a bit buggy

**Testing**
I you want to test the first alpha version (v2.0.0-alpha01), you can download the attached APK file now and install it on your device. You may have to enable an option in your device's settings with a name like "Allow from unknown sources" or if you have a newer Android version there might be a popup with a toggle "Allow from this source".
You can install the APK over your current installation, because it has the same signature.
After update, some or all app preferences may have the default status because their identifiers changed internally and I won't write migrations for them.

Please report any bug or misbehavior in a [new issue](https://github.com/patzly/grocy-android/issues) **if there isn't already an open one**.
New or rewritten pages may also contain issues which were fixed in previous releases (so they may have a closed issue here) – please report them too in a new issue.

If you just want to say thanks for our work and this new version or motivate us, you can write a message in the new Matrix room [#grocy-android:matrix.org](https://element.io/get-started) .
Thanks!

**Translating**
There are lots of new string to translate, so you can start translating them on [Transifex](https://www.transifex.com/grocy-android/grocy-android). :)

---
Sorry to the guys whose comments I marked as off-topic: It was not against you, but I did it because else maybe more people would answer the same and spam email notifications to everyone like @towo wrote.

And sorry, our English is not *brilliant* yet... If your mother tongue is English and you find language mistakes in the app, please report them also in the issue tracker.

And finally, sorry for the long waiting time – I'm studying computer science since a few months now and hence I don't have that much time anymore to work on this project and my brother @patzly also has a lot to do.

PS: Please click on the reaction buttons to react to a comment, instead of posting new comments here.

Best regards
Dominic & Patrick


APK: [Grocy-Android-v2.0.0-alpha01.zip](https://github.com/patzly/grocy-android/files/5946299/Grocy-Android-v2.0.0-alpha01.zip) (you have to unzip this file)

**Comment by samnewman86** [comment]
Thanks for your hard work and uploading the alpha app.
I've added the first issue for you..😉

**Comment by patzly** [comment]
Hey guys, new alpha is ready!

Changelog:
- **New:** Language option in appearance settings (#322)
- **Fixed:** Update info wrongly shown (#322)
- **Fixed:** Crash on quantity units list refresh (#331)
- **Fixed:** Wrong product name after adding new product with "&" in its name (#333)
- **Fixed:** Wrong resource qualifier for Hebrew
- Updated translations

APK: [Grocy-Android-v2.0.0-alpha02.zip](https://github.com/patzly/grocy-android/files/5961616/Grocy-Android-v2.0.0-alpha02.zip) (you have to unzip this file)

**Please don't post any comments here anymore so that not all others who have subscribed to this issue are notified!**

**Comment by patzly** [comment]
New alpha is ready!

Changelog:
- **New:** Official support for Home Assistant grocy add-on (#249)
- **New:** App can handle `username:password` in server URL for HTTP authentication (#273)
- **Improved:** Rewritten login flow
- **Improved:** Update certificate compatibility information
- **Improved:** Demo server changes when the language is changed (#346)
- **Improved:** Price and store lines on purchase page confirmation bottom sheet
- **Improved:** Updated translations
- **Fixed:** Crash when open store selection on purchase page (#347)
- **Fixed:** Crash on purchase page when price was empty & scan mode active
- **Fixed:** Behavior on purchase & consume pages when quantity units of product were missing
- **Fixed:** Crash on consume page when location could not be found
- **Fixed:** Crash on consume page when `location_is_freezer` was empty

APK: [Grocy-Android-v2.0.0-alpha03.zip](https://github.com/patzly/grocy-android/files/5997747/Grocy-Android-v2.0.0-alpha03.zip) (you have to unzip this file)

Best regards!
Dominic and Patrick

**Comment by patzly** [comment]
## Alpha 4 is out!

- **New:** Animated splash screen and logo
- **Improved:** Keyboard does't hide save/clear buttons on text input sheet anymore (#261)
- **Improved:** Added help texts to login form (shown with Home Assistant add-on)
- **Improved:** Updated translations
- **Fixed:** RuntimeExceptions in log if Home Assistant grocy add-on is not used (#273)
- **Fixed:** Mixed languages if activity is restarted (#260)
- **Fixed:** Language variants not working (#264)
- **Fixed:** Updates to master data were pushed to the wrong server (#363)
- **Fixed:** Login with Home Assistant grocy add-on using Nabu Casa service (#249)
- **Fixed:** Save bottom notes on shopping mode page not working (#362)
- **Fixed:** Updated bottom notes not displayed in shopping list

APK: [Grocy-Android-v2.0.0-alpha04.zip](https://github.com/patzly/grocy-android/files/6016830/Grocy-Android-v2.0.0-alpha04.zip) (you have to unzip this file)

Cheers! 🥂

**Comment by patzly** [comment]
## Alpha 5 is out!
- Improved: Language selection and behavior
- Improved: Display exact error message if there was an AuthFailureError at login (#249)
- Improved: Updated translations
- Fixed: Wrong factor for unit multiplies on shopping list item edit page (#356)
- Fixed: Wrong language selected if code contains specific region (#372)
- Fixed: Language change does not effect shortcuts (#373)

APK: [Grocy-Android-v2.0.0-alpha05.zip](https://github.com/patzly/grocy-android/files/6058273/Grocy-Android-v2.0.0-alpha05.zip) (you have to unzip this file)

**Comment by larsreinhardt** [comment]
Im in

**Comment by patzly** [comment]
## Alpha 6 is out!

Hey guys! After a few months of lacking time for this project due to study and work, we are happy to release the next (and hopefully one of the last) alpha version. Thanks to everyone for testing and reporting!

Changes in this release include:

- **New:** Stock overview page now has list animations and offline loading
- **New:** _Scan mode_ is now _Quick mode_ and has a better help text (#332)
- **New:** Jost* typeface by Owen Earl (we hope you like it)
- **New:** Barcode search works on new stock overview page (#375)
- **Improved:** Stock information on start page
- **Improved:** Lock current screen orientation while camera scanner frame is visible (#358)
- **Improved:** Camera scanner frame now has a fixed width (#334)
- **Improved:** Stock overview page shortcut is now available
- **Improved:** Swipe behavior is much better
- **Improved:** Updated translations
- **Fixed:** Filter didn't work on master data products page (#370)
- **Fixed:** Don't show inactive products on consume & purchase page (#382)
- **Fixed:** _Add_ button was gone in shopping lists bottom sheet
- **Fixed:** Location filter in new stock overview now behaves like on server (#248)
- **Fixed:** Products are hidden in new stock overview if "hide on stock" is active
- **Fixed:** Quick consume & open work again in stock overview (#386)
- **Fixed:** Amount which is set in product as quick consume amount is used (#359)

APK: [Grocy-Android-v2.0.0-alpha06.zip](https://github.com/patzly/grocy-android/files/6377908/Grocy-Android-v2.0.0-alpha06.zip) (you have to unzip this file)

**Comment by patzly** [comment]
## Alpha 7 is out!

Changes in this release include:

- **New:** Transfer page and inventory page (finally!) (#385 and #115)
- **New:**  We use Conscrypt now to support TLS1.3 on older Android versions (#163)
- **New:**  You can duplicate products now (#146)
- **Improved:** List sorting uses rules of currently selected language (#371)
- **Improved:** You can open purchase or consume page from stock for one product now
- **Improved:** Long-tap on shopping list card on start page to open shopping mode
- **Improved:** Shopping list item edit page has the camera scanner frame now
- **Improved:** Up and down page animations
- **Improved:** Focus product field after unknown barcode was scanned
- **Improved:** Amount field in forms has a clear button which also opens keyboard
- **Improved:** Menu in product overview bottom sheet has a new submenu
- **Improved:** Updated translations
- **Fixed:** Swipe buttons in stock overview didn't work for last two items
- **Fixed:** Menu items on bottom were not clickable sometimes (#383)
- **Fixed:** Parent product field is locked if product is already parent product (#349)
- **Fixed:** Barcode chips have not cleared the barcode when removed
- **Fixed:** Quick mode on consume page had wrong initial state sometimes
- **Fixed:** Remove barcode from forms during 'clear form' action

APK: [Grocy-Android-v2.0.0-alpha07.zip](https://github.com/patzly/grocy-android/files/6412380/Grocy-Android-v2.0.0-alpha07.zip) (you have to unzip this file)

Join our Matrix room for feedback and general questions: [#grocy-android:matrix.org](https://element.io/get-started)

**Comment by patzly** [comment]
## Beta 1 is out!

You can download the APK on the [releases page](https://github.com/patzly/grocy-android/releases/tag/v2.0.0-beta1).

**Comment by patzly** [comment]
### Hey guys!
We would like to thank you for the intensive testing and the many issues you reported.
With today's release of the first beta on Google Play, we are now closing this issue. To be notified about new releases, you can watch our GitHub project and select "Releases".

See you then!
